### ISSUE TITLE
subscriptions not loading on startup

### ISSUE THREAD

**Issue by grrrrr** [FreeTubeApp/FreeTube#552]
**Behavior of the program**

On launching FreeTube 0.8.0,  the program fails to load/refresh the subscriptions.  Instead it displays the following error on the subscription page until you manually refresh

`Your Subscription list is currently empty. Start adding subscriptions to see them here.`

Subscriptions can be seen in the sidebar

**Expected behavior**

Opening FreeTube, should display the previously fetches videos or force a refresh of the subscription list

**To Reproduce**

Please add all steps to reproduce the behavior:
1. Launch FreeTube
2. Observe message about not having subscriptions
3. refresh subscriptions
4. subscription videos appear

**Desktop (please complete the following information):**

 - OS: Debian
 - OS Version: 10.6
 - FreeTube version: 0.8.0

**Comment by GilgusMaximus** [comment]
Yes. That issue can occur sometimes. A bit of investigation will be needed in order to find the cause.

**Comment by deejf** [comment]
This issue occurs on every startup for me. I assumed it was because I switched my subscription pulls to RSS, for whatever reason.

OS: Fedora
OS version: 28 (yes, I'm way out of date)
Freetube version: 0.9.1

**Comment by GilgusMaximus** [comment]
Actually. This is intended behavior as I have learned. Because of the rate limiting problem, the page does not auto load when you open your subscriptions. At least no always, it still seems to be a bit inconsistent sometimes.

But this could be implemented as a setting, so that you can enable auto fetching

**Comment by grrrrr** [comment]
@GilgusMaximus I think the confusing part is that it says you have no subscriptions.   Maybe the better option would be to display a message like `refresh to update subscriptions`    It's much less of a confusing than seeing a message that you do not have subscriptions, which admittedly the first time I saw it I briefly though FreeTube lost everything

**Comment by GilgusMaximus** [comment]
That is true. It would probably be best to implement a storage method, which saves the first page of the videos on closing and then loads them again on opening.

**Comment by deejf** [comment]
The other irritating part is, why can't it load the stuff it had already loaded previously? If I open it every day, then there were videos loaded yesterday that, intuitively, the program *should* remember. But doesn't.

**Comment by GilgusMaximus** [comment]
@deejf As I said in the comment above yours, this could be implemented. We are open For PR's, as this is rather easy to do

**Comment by deejf** [comment]
@GilgusMaximus It might be easy to do, except the last coding I ever did was in Pascal in 1994. I'm an author and editor, not a programmer, so it is not easy for _me_ to do, since I don't even have the foundational knowledge to begin to learn the easy stuff.

**Comment by GilgusMaximus** [comment]
I did not experience this issue anymore in recent weeks. Anyone with the newest version and still experiencing this? if not, I would close this issue

**Comment by lfom** [comment]
Bump. Still happens here (v0.12.0 Beta on Pop!_OS 20.10). I did not import anything, I have only added subscriptions inside Freetube and I have two profiles. After a cold start, Freetube always shows:
```
Your Subscription list is currently empty. Start adding subscriptions to see them here.
```
what is very misleading. If I open another section and go back or refresh the page, then it shows the videos normally.

**Comment by Svallinn** [comment]
This is most likely fixed with my commit, but I can never be 100% sure since it happens too arbitrarily for me.
If it does seem like it still happens in the latest automated build, please don't comment in here and open a new issue.

**Comment by lfom** [comment]
Just a simple suggestion: modify the error message, say something like "Subscriptions not loaded" instead of saying that it empty since it isn't. This is the biggest issue IMHO.

**Comment by Svallinn** [comment]
That shouldn't be necessary, since they should now always automatically load on startup, so that message will only truly appear if have no subscriptions.

**Comment by grrrrr** [comment]
@svallin,  as you said your self you are not 100% sure it's fixed,  so would it not make sense to account for the potential failure and make that error message usefully descriptive?

**Comment by Svallinn** [comment]
Let me provide some extra context then.
I used a build that included the logic that fixes the subscription loading issue for **a week and a half** and I'm quite a frequent user.
I have seen this issue before so I know what it looks like; however, it did not occur a single time during that week and a half for me.
So, while I do say I'm not 100% sure about it being fixed, I can say I'm 99% sure of it.
The only reason I wrote that comment is purely because, in the very unlikely chance that I was just extremely lucky, I just want to cover all bases.
